### PR TITLE
include linter version in output analysis_options.yaml

### PR DESCRIPTION
**Pull request by robbecker-wf**
# Overview
Fixes https://github.com/Workiva/abide/issues/6
Users would like to know which version of the linter package was used to update their `analysis_options.yaml` and if it is up to date with the latest.

# Changes
`abide_upgrade` will include the linter version used in the generated `lib/abide.yaml` under a `__linter_version` key. This is used when users do an `abide update`. It now includes the version in a comment at the top of the file and a link to the linter package to easily check for the latest version.

# Note
Having an extra top level key in the abide.yaml sort of conflicts with the assumption that each top level key is a lint rule name. In a few places, the top level keys are iterated through with that assumption. To work around this, a new function was added that gets the top level keys from a yaml file and excludes any keys starting with double underscore `__`. 

# Testing
 - CI passes
 - Run an abide update with this PR branch installed and you should get the linter version in your analysys_options.yaml

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/abide.yaml [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
-# 
-# recommendations: 
+#
+# recommendations:
 #  required    - must be present in analysis_options.
 #  recommended - optional but recommended. Add it as you have time.
 #  optional    - optional. Up to you and your team to decide.
 #  avoid       - should not be present in analysis_options. Its presence is an error.
 
+__linter_version: 0.1.82
```
Comment (robbecker-wf) [inline lib/abide.yaml]: New

File: lib/abide.yaml [hunk 2]
```diff
@@ -551,175 +553,175 @@ one_member_abstracts:
...
 
 prefer_bool_in_asserts:
   description: "Prefer using a boolean as the assert condition."
   maturity: deprecated
   group: style
   docs: http://dart-lang.github.io/linter/lints/prefer_bool_in_asserts.html
-  recommendation: optional
-  reason: 
+  recommendation: avoid
+  reason: This lint rule has been deprecated
```
Comment (robbecker-wf) [inline lib/abide.yaml]: changed recommendation to `avoid` since it is deprecated

File: lib/abide.yaml [hunk 3]
```diff
@@ -855,63 +865,63 @@ recursive_getters:
...
 super_goes_last:
   description: "Place the `super` call last in a constructor initialization list."
-  maturity: stable
+  maturity: deprecated
   group: style
   docs: http://dart-lang.github.io/linter/lints/super_goes_last.html
-  recommendation: optional
-  reason: 
+  recommendation: avoid
+  reason: This lint rule has been deprecated
```
Comment (robbecker-wf) [inline lib/abide.yaml]: changed recommendation to `avoid` since it is deprecated

**Comment by robbecker-wf** [comment]
Verified it works in another project
image

File: Dockerfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
-FROM google/dart:2.0.0 as build
+FROM google/dart:2.1.1 as build
```
Comment (georgelesica-wf) [inline Dockerfile]: Should we also bump the version in the pubspec?

**Comment by georgelesica-wf** [comment]
+1

File: Dockerfile (hunk 4 above)
Comment (robbecker-wf) [reply Dockerfile]: The current range is `'>=1.23.0 <3.0.0'` so I don't think we need to update it.

**Comment by robbecker-wf** [comment]
@Workiva/release-management-p

**Pull request merged**